### DISCUSSION TITLE
Multiple instances of BICUBIC backend

### DISCUSSION THREAD

**Discussion by eliesaikali** [CoolProp/CoolProp#2240] [Q&A]
Hello and sorry for all these questions lately (I am new with CoolProp and I am trying to use it in a complex multiphase application).

I notice that ther is (maybe) a bug with creating multiple instances of BICUBIC backend ?
In fact, in my application, I create several instances of AbstractState::factory (an instance by fluid ...)
I am having issues when defining more than one instance of  BICUBIC backend (the constructor forces the second phase to unknown) and the error is a bit weird : surface tension is only defined within the two-phase region; Try PQ or QT inputs.

I just created a simple main to reproduce the issue ! 
thanks for your help

PS : no problem if I copy the pointer ! 

===========================
#include "AbstractState.h"
#include "CoolProp.h"
#include <iostream>

using namespace CoolProp;

int main(int argc, char const *argv[])
{
  CoolProp::AbstractState *fluide1 = CoolProp::AbstractState::factory("BICUBIC&HEOS", "Water");
  CoolProp::AbstractState *fluide2 = CoolProp::AbstractState::factory("BICUBIC&HEOS", "Water");

  for (int i = 0; i < 3; i++)
    {
      fluide1->update(CoolProp::PQ_INPUTS, 1.e+5, 1); // SI units
      fluide2->update(CoolProp::PQ_INPUTS, 1.e+5, 1); // SI units
      //  fluide1->specify_phase(CoolProp::iphase_gas);
      //  fluide2->specify_phase(CoolProp::iphase_gas);
      std::cout << "SIGMA: " << fluide1->surface_tension() << std::endl;
      std::cout << "SIGMA: " << fluide2->surface_tension() << std::endl;
      std::cout << std::endl;
    }

  delete fluide1;
  delete fluide2;
  return 1;
}
===========================================

**Comment by ibell** [comment]
Why you can't have two instances of the same BICUBIC with the same fluid I don't know, does it work for you if one is water and the other propane?

Unrelated: don't (never!) use bare pointers; initialize them immediately into shared pointers or unique pointers so they are automatically deleted

**Reply by eliesaikali** [reply]
Thanks for your reply !
Same error with different fluids, already tried... it's an issue with the backend BICUBIC.

And thanks for the remark, you are right, should always use smart pointers !

**Reply by eliesaikali** [reply]
I think that its just an issue with the surface_tension method (implemented in helmhotzdmixture backend). The enum of the phase is set to unknown in the second instance. Set phase method does nothing too ...

**Reply by havifloja** [reply]
I had the same problem. The surface_tension implementation in tabular backend uses the original function (without tables). It seems there is a bug here. Both clearing of the state and setting the phase is missing. I could avoid the error by starting the calc_surface_tension method in tabular backends with the following two lines:
```
this->AS->clear();
this->AS->specify_phase(iphase_twophase);
```

**Reply by ibell** [reply]
It would be great if someone could look into the C++ and fix this. It seems like a beginner-friendly bugfix
